rosbridge_library/src/rosbridge_library/internal/services.py

```diff
@@ -120,9 +122,15 @@ def call_service(node_handle, service, args=None):
     # Populate the instance with the provided args
     args_to_service_request_instance(service, inst, args)
 
-    client = node_handle.create_client(service_class, service)
+    client_key = f"{service}:{service_type}"
+    if client_key in clients:
+        client = clients[client_key]
+    else:
+        client = node_handle.create_client(service_class, service)
+        clients[client_key] = client
 
     result = client.call(inst)
+
```

Comment:
```suggestion
    node_handle.destroy_client(client)
```

Reply:
@achim-k

No.
> And I know that node_handle.destroy_client(client) does not free all memory.

Adding this code, memory leak issue is not solved.

Reply:
My first commit was this, but I found it didn't release all the memory.

Reply:
@achim-k 

releasing resources by `node_handle.destroy_client(client)` is rclpy responsibility, so I'll change this PR, and I'll also submit to fix this issue to rclpy repo.

Reply:
If you create a `rclpy` issue / PR, could you add a link here? Thanks!